File: java/src/main/java/deltix/luminary/implementation/DecoratorValueNode.java
package deltix.luminary.implementation;

import deltix.luminary.NameValuePair;
import org.jetbrains.annotations.NotNull;

import java.util.ArrayList;
import java.util.List;

public class DecoratorValueNode {
    private final String name;
    private final List<NameValuePair> arguments;

    public DecoratorValueNode(String name, List<NameValuePair> arguments) {
        this.name = name;
        this.arguments = arguments != null ? arguments : new ArrayList<NameValuePair>();
    }

    public String getName() {
        return name;
    }

    @NotNull
    public List<NameValuePair> getArguments() {
        return arguments;
    }
}

File: csharp/main/Implementation/DecoratorValueNode.cs
using System;
using System.Collections.Generic;
using JetBrains.Annotations;

namespace Deltix.Luminary.Implementation
{
	public class DecoratorValueNode
	{
		public DecoratorValueNode([NotNull] String name, [CanBeNull, ItemNotNull] List<NameValuePair> arguments)
		{
			Name = name;
			Arguments = arguments ?? new List<NameValuePair>();
		}

		[NotNull]
		public String Name { get; }

		[NotNull, ItemNotNull]
		public List<NameValuePair> Arguments { get; }
	}
}
